File: backend/src/main/java/br/ufpr/sept/so2/modules/bff/application/ports/EventosDashboardQueryPort.java
package br.ufpr.sept.so2.modules.bff.application.ports;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

public interface EventosDashboardQueryPort {

    EventosDashboard consultar(OffsetDateTime agora);

    record EventosDashboard(int hoje, List<EventoResumo> proximos) {
    }

    record EventoResumo(
            UUID id,
            String titulo,
            OffsetDateTime inicioEm,
            OffsetDateTime fimEm,
            boolean janelaAtiva
    ) {
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/bff/application/ports/EventosDashboardQueryPort.kt
package br.ufpr.sept.so2.modules.bff.application.ports

import java.time.OffsetDateTime
import java.util.UUID

fun interface EventosDashboardQueryPort {
    fun consultar(agora: OffsetDateTime): EventosDashboard

    data class EventosDashboard(val hoje: Int, val proximos: List<EventoResumo>)

    data class EventoResumo(
        val id: UUID,
        val titulo: String,
        val inicioEm: OffsetDateTime,
        val fimEm: OffsetDateTime,
        val janelaAtiva: Boolean,
    )
}
